flytekit/core/map_task.py

```diff
@@ -1,72 +1,168 @@
-from typing import Any, Optional
+import os
+from typing import Any, Dict, List, Optional, Type
 
+from flytekit.common.tasks.raw_container import _get_container_definition
 from flytekit.core.base_task import PythonTask, TaskMetadata
+from flytekit.core.context_manager import ExecutionState, FlyteContext, SerializationSettings
 from flytekit.core.interface import transform_interface_to_list_interface
+from flytekit.core.python_function_task import PythonFunctionTask, get_registerable_container_image
+from flytekit.models.interface import Variable
+from flytekit.models.task import Container
 
 
 class MapPythonTask(PythonTask):
     """
-    TODO We might need a special entrypoint to start execution of this task type as there is possibly no instance of this
-    type and it needs to be dynamically generated at runtime. We can easily generate it by passing it the actual task
-    that is to be generated.
-
-    To do this we might have to give up on supporting lambda functions initially
+    TODO: support lambda functions
     """
 
-    def __init__(self, tk: PythonTask, metadata: Optional[TaskMetadata] = None, **kwargs):
+    def __init__(
+        self,
+        tk: PythonFunctionTask,
+        metadata: Optional[TaskMetadata] = None,
```

Comment:
I think you can skip the taskmetadata right? that is in the base task so will be auto in **kwargs?